tests/base-jupyter/test_user_options.py

```diff
@@ -0,0 +1,290 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+import logging
+import pathlib
+import time
+
+import pytest  # type: ignore
+
+from tests.conftest import TrackedContainer
+
+LOGGER = logging.getLogger(__name__)
+
+
+def test_uid_change(container: TrackedContainer) -> None:
+    """Container should change the UID of the default user."""
+    logs = container.run_and_wait(
+        timeout=120,  # usermod is slow so give it some time
+        tty=True,
+        user="root",
+        environment=["NB_UID=1010"],
+        command=["start.sh", "bash", "-c", "id && touch /opt/conda/test-file"],
+    )
+    assert "uid=1010(jovyan)" in logs
+
+
+def test_gid_change(container: TrackedContainer) -> None:
+    """Container should change the GID of the default user."""
+    logs = container.run_and_wait(
+        timeout=10,
+        tty=True,
+        user="root",
+        environment=["NB_GID=110"],
+        command=["start.sh", "id"],
+    )
+    assert "gid=110(jovyan)" in logs
+    assert "groups=110(jovyan),100(users)" in logs
+
+
+def test_nb_user_change(container: TrackedContainer) -> None:
+    """Container should change the username (`NB_USER`) of the default user."""
+    nb_user = "nayvoj"
+    running_container = container.run_detached(
+        tty=True,
+        user="root",
+        environment=[f"NB_USER={nb_user}", "CHOWN_HOME=yes"],
+        command=["start.sh", "bash", "-c", "sleep infinity"],
+    )
+
+    # Give the chown time to complete.
+    # Use sleep, not wait, because the container sleeps forever.
+    time.sleep(1)
+    LOGGER.info(f"Checking if the user is changed to {nb_user} by the start script ...")
+    output = running_container.logs().decode("utf-8")
+    assert "ERROR" not in output
+    assert "WARNING" not in output
+    assert (
+        f"username: jovyan       -> {nb_user}" in output
+    ), f"User is not changed to {nb_user}"
+
+    LOGGER.info(f"Checking {nb_user} id ...")
+    command = "id"
+    expected_output = f"uid=1000({nb_user}) gid=100(users) groups=100(users)"
+    cmd = running_container.exec_run(command, user=nb_user, workdir=f"/home/{nb_user}")
+    output = cmd.output.decode("utf-8").strip("\n")
+    assert output == expected_output, f"Bad user {output}, expected {expected_output}"
+
+    LOGGER.info(f"Checking if {nb_user} owns his home folder ...")
+    command = f'stat -c "%U %G" /home/{nb_user}/'
+    expected_output = f"{nb_user} users"
+    cmd = running_container.exec_run(command, workdir=f"/home/{nb_user}")
+    output = cmd.output.decode("utf-8").strip("\n")
+    assert (
+        output == expected_output
+    ), f"Bad owner for the {nb_user} home folder {output}, expected {expected_output}"
+
+    LOGGER.info(
+        f"Checking if home folder of {nb_user} contains the hidden 'work' folder with appropriate permissions ..."
```

Comment:
The `work` folder is not hidden, and we should check that hidden folder are copied properly.

Reply:
There are hidden _files_, but `work` is the only directory in the home folder:
```bash
$ docker run -it --rm jupyter/base-jupyter ls -la  
total 32
drwsrws--- 1 jovyan users 4096 Nov  2 13:54 .
drwxr-xr-x 1 root   root  4096 Nov  1 19:35 ..
-rw-rw-r-- 1 jovyan users  220 Jan  6  2022 .bash_logout
-rw-rw-r-- 1 jovyan users 3823 Nov  1 19:35 .bashrc
-rw-rw-r-- 1 jovyan users  807 Jan  6  2022 .profile
-rw-rw-r-- 1 jovyan users  171 Nov  2 13:54 .wget-hsts
drwsrwsr-x 2 jovyan users 4096 Nov  1 19:35 work
```
I will update the doc to remove "hidden", but will need your advice otherwise.